DMCompiler/Compiler/DM/DMParser.cs

```diff
@@ -1951,120 +1956,140 @@ public DMASTProcStatementLabel Label(DMASTIdentifier expression)
-                                        break;
-                                    default: //Unimplemented escape sequence
+                                    } else if (escapeSequence.StartsWith("n")) {
+                                        stringBuilder.Append('\n');
+                                        stringBuilder.Append(escapeSequence.Skip(1).ToArray());
+                                    } else if (escapeSequence.StartsWith("t")) {
+                                        stringBuilder.Append('\t');
+                                        stringBuilder.Append(escapeSequence.Skip(1).ToArray());
+                                    } else if (escapeSequence == "ref") {
+                                        currentInterpolationType = StringFormatTypes.Ref;
+                                    } else {
                                         Error("Invalid escape sequence \"\\" + escapeSequence + "\"");
-                                        break;
+                                    }
+                                } else
+                                {
+                                    escapeSequence += c;
+                                    switch (escapeSequence)
+                                    {
+                                        case "[":
+                                        case "]":
+                                        case "<":
+                                        case ">":
+                                        case "\"":
+                                        case "'":
+                                        case "\\":
+                                        case " ":
+                                        case ".":
+                                            stringBuilder.Append(escapeSequence);
+                                            break;
+                                        default: //Unimplemented escape sequence
+                                            Error("Invalid escape sequence \"\\" + escapeSequence + "\"");
+                                            break;
+                                    }
                                 }
+
+                                break;
+                            }
+                            default:
+                            {
+                                if (bracketNesting == 0) {
+                                    stringBuilder.Append(c);
+                                }
+
+                                break;
                             }
-                        } else if (bracketNesting == 0) {
-                            stringBuilder.Append(c);
                         }
                     }
 
                     if (bracketNesting > 0) Error("Expected ']'");
 
                     string stringValue = stringBuilder.ToString();
-                    if (interpolationValues.Count == 0) {
+                    if (interpolationValues is null || interpolationValues.Count == 0) {
```

Comment:
As long as `interpolationValues` isn't null it should have at least 1 value

```suggestion
                    if (interpolationValues is null) {
```